app/javascript/helpers/validate-names.js

```diff
@@ -0,0 +1,7 @@
+const validateName = (target, name) => API.get(`/api/${target}?expand=resources&attributes=name`)
+  .then(({ resources }) => resources.map((resource) => resource.name))
+  .then((results) => {
+    return results.includes(name) ? `The name "${name}" already exists in ${target}` : undefined;
+  });
+
+export default validateName;
```

Comment:
this can be written like-

```
const validateName = (target, name) => API.get(`/api/${target}?expand=resources&attributes=name`)
  .then(({ resources }) => resources.map((resource) => resource.name))
  .then((results) => (results.includes(name) ? `The name "${name}" already exists in ${target}` : undefined));

export default validateName;
```

Reply:
just out of curiosity: the `sprintf` is required because the string formatting is not safe?